client/packages/programs/src/JsonForms/components/Note.tsx

```diff
@@ -0,0 +1,172 @@
+  const [localAuthor, setLocalAuthor] = useState<string | undefined>(
+    data?.authorName ??
+      // TO-DO: Use full name for default once available in database
+      config.user?.name
+  );
+  // timestamp of the last key stroke
+  const [latestKey, setLatestKey] = useState<number>(0);
+  const { localisedDateTime } = useFormatDateTime();
+
+  const { errors: zErrors, options: schemaOptions } = useZodOptionsValidation(
+    Options,
+    uischema.options
+  );
+
+  const error = !!errors || !!zErrors;
+
+  const onChange = (
+    text: string | undefined,
+    authorName: string | undefined
+  ) => {
+    const authorId = config.user?.id;
+    const created = data?.created ?? new Date().toISOString();
+    handleChange(
+      path,
+      error
+        ? undefined
+        : {
+            text,
+            authorId,
+            authorName,
+            created,
+          }
+    );
+  };
+
+  // debounce avoid rerendering the form on every key stroke which becomes a performance issue
+  const onChangeText = useDebounceCallback(
+    (value: string) => value && onChange(value, localAuthor),
+    [path, error, localAuthor]
+  );
+
+  const onChangeAuthor = useDebounceCallback(
+    (value: string) => value && onChange(localText, value),
+    [path, error, localText]
+  );
+
+  const t = useTranslation('programs');
+
+  const helperText = zErrors ?? errors;
+
+  useEffect(() => {
+    // Using debounce, the actual data is set after 500ms after the last key
+    // stroke (localDataTime). If data is set from the outside, e.g. through a
+    // reset, we want to update our local data as well. To distinguish between
+    // debounced events and external data updates we only take data that comes
+    // in at least 500ms after the last key stroke, i.e. it must be set from the
+    // outside.
+    if (Date.now() > latestKey + 500) {
+      setLocalText(data?.text);
+    }
+  }, [data]);
+
+  if (!props.visible) {
+    return null;
+  }
+
+  const width = schemaOptions?.width ?? '100%';
+  const rows = schemaOptions?.rows ?? 5;
+
+  const authorStyle = {
+    textAlign: 'right',
+    fontSize: '90%',
+    color: 'gray.dark',
+  };
+
+  return enabled ? (
+    <div>
+      <DetailInputWithLabelRow
+        label={t('control.note.text-label')}
+        inputProps={{
+          value: localText ?? '',
+          name: 'text',
+          sx: { margin: 0.5, width },
+          onChange: e => {
+            setLatestKey(Date.now());
+            setLocalText(e.target.value);
+            onChangeText(e.target.value);
+          },
+          disabled: !props.enabled,
+          error,
+          helperText,
+          FormHelperTextProps: error
+            ? { sx: { color: 'error.main' } }
+            : undefined,
+          required: props.required,
+          multiline: true,
+          rows,
+        }}
+        labelWidthPercentage={FORM_LABEL_WIDTH}
+        inputAlignment={'start'}
+      />
+      <DetailInputWithLabelRow
```

Comment:
I don't think that the author should be an editable field, you could put anyone's name in here! Makes more sense to tie the author to the logged in user and simply have this as read-only text.

Reply:
Done in latest changes. 👍